File: C/29.c
// 29.Write a program to find all pairs in an
// array whose sum is equal to a given number.
//Example 1: Input: nums = [2,7,11,15], target = 9 , Output:
// [0,1].



int main(){
	
	int arr[]={2,7,11,15};
	int len = sizeof(arr)/sizeof(arr[0]);
	int target;
	int found=0;
	int ans[10];
	printf("enter your target : ");
	scanf("%d",&target);
	int i,j;
	for(i=0;i<len;i++){
		for(j=i+1;j<len;j++){
			if(arr[i] + arr[j] == target){
			   printf("Output: [%d, %d]\n", i, j);
			   found++;
			}
		}
	}
	if(found==0){
		printf("pairs not found");
	}
	return 0;
}

File: JS/29.js
function target(num){
    if(typeof num !== 'number' ){
        console.log("this is not number")
        return;
    }
    let ans =[];
    let arr =  [2,7,11,15]
    for(i=0;i<arr.length;i++){
        for(j=i+1;j<arr.length;j++){
            if(arr[i]+arr[j]==num){
                ans.push([arr[i],arr[j]])
            }
        }
    }
    if(ans.length>0){
        console.log(ans)
    }else{
        console.log("your num pairs are not in array.")
    }
}
target(9)